src/rscminus/common/ChatCipher.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * rscminus
+ *
+ * This file is part of rscminus.
+ *
+ * rscminus is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * rscminus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with rscminus. If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ * Authors: see <https://github.com/OrN/rscminus>
+ */
+
+package rscminus.common;
+
+import rscminus.game.NetworkStream;
+import rscminus.game.entity.player.ChatMessage;
+
+import java.util.HashMap;
+
+public class ChatCipher {
+
+    /**
+     * Unicode characters that are mapped to specific byte values
+     * The array is for byte->char, the dict is for char->byte
+     * Note: We found no evidence of these unicode characters in scraped replays
+     * so we can't be sure this was actually possible. But it seems highly likely that it was.
```

Comment:
clarify "this" as "we can't be sure that the server would actually transmit these mapper characters [...]"